javascript/signFunction.js

```diff
@@ -1,59 +1,62 @@
-const inputFunction = {
-  // 이메일 입력란에 focus 일 때 이벤트 핸들러
-  emailFocus: (e) => {
-    e.target.classList.remove('input-error')
-    e.target.parentElement.lastElementChild.textContent = ''
-  },
+import { emailDoubleCheck } from "./api/auth/emailCheck.js"
+import { emailRegex, passwordRegex } from "./regex.js"
 
-  // 이메일 입력란에서 blur 일 때 이벤트 핸들러
-  emailBlur: (e, emailForm, accountInfo, inputEmail, signupValidation = false) => {
-    const matchingAccountEmail = accountInfo.find((item) => item.email === inputEmail?.value)
-  
-    if (e.target.value === '') {
-      e.target.classList.add('input-error')
-      e.target.parentElement.lastElementChild.textContent = '이메일을 입력해 주세요.'
-    }
-    else if (!e.target.value.match(emailForm)) {
-      e.target.classList.add('input-error')
-      e.target.parentElement.lastElementChild.textContent = '올바른 이메일 주소가 아닙니다.'
-    }
-    else if (signupValidation && e.target.value === matchingAccountEmail?.email) {
-      e.target.classList.add('input-error')
-      e.target.parentElement.lastElementChild.textContent = '이미 사용 중인 이메일입니다.'
-    }
-  },
+// 이메일 입력란에서 focus 일 때 이벤트 핸들러
+const emailFocus = (inputEmail, errorMessage) => {
+  inputEmail.classList.remove('input-error')
+  errorMessage.textContent = ''
+}
+
+// 이메일 입력란에서 blur 일 때 이벤트 핸들러
+const emailBlur = (inputEmail, errorMessage, signupValidation = false) => {
+  if (inputEmail.value === '') {
+    inputEmail.classList.add('input-error')
+    errorMessage.textContent = '이메일을 입력해 주세요.'
+    return
+  }
+  if (!inputEmail.value.match(emailRegex)) {
+    inputEmail.classList.add('input-error')
+    errorMessage.textContent = '올바른 이메일 주소가 아닙니다.'
+    return
+  }
+  if (signupValidation) {
+    emailDoubleCheck()
+    return
+  }
```

Comment:
하단에 추가로직이 없으니 return문은 불필요할 것 같습니다.
```suggestion
  if (signupValidation) {
    emailDoubleCheck()
  }
```

Reply:
네!! 알겠습니다.